plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Instance/Eq.hs

```diff
@@ -19,11 +20,23 @@ import PlutusCore.Rename.Monad
 import Universe
 
 import Data.Foldable (for_)
+import Data.Hashable
 
 instance (GEq uni, Closed uni, uni `Everywhere` Eq, Eq fun, Eq ann) =>
             Eq (Term Name uni fun ann) where
     term1 == term2 = runEqRename $ eqTermM term1 term2
 
+type HashableTermConstraints uni fun ann =
+  ( GEq uni
+  , Closed uni
+  , uni `Everywhere` Eq
+  , uni `Everywhere` Hashable
+  , Hashable ann
+  , Hashable fun
+  )
+
+instance HashableTermConstraints uni fun ann => Hashable (Term Name uni fun ann)
```

Comment:
Why don't we just have a `Hashable name` constraint and have it work for any `name`?